From akpm@osdl.org Mon Aug 14 23:11:35 2006
Message-Id: <200608150611.k7F6B9Bi013394@shell0.pdx.osdl.net>
From: Andrew Morton <akpm@osdl.org>
Subject: USB: rtl8150_disconnect() needs tasklet_kill()
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net, akpm@osdl.org, miles.lane@gmail.com,
   petkan@users.sourceforge.net
From: akpm@osdl.org
Date: Mon, 14 Aug 2006 23:11:09 -0700

From: Andrew Morton <akpm@osdl.org>

We need to wait until any currently-running handler has completed.  Fixes an
unplug-time oops reported by "Miles Lane" <miles.lane@gmail.com>.

Cc: "Petko Manolov" <petkan@users.sourceforge.net>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/net/rtl8150.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/usb/net/rtl8150.c
+++ gregkh-2.6/drivers/usb/net/rtl8150.c
@@ -972,6 +972,7 @@ static void rtl8150_disconnect(struct us
 	if (dev) {
 		set_bit(RTL8150_UNPLUG, &dev->flags);
 		tasklet_disable(&dev->tl);
+		tasklet_kill(&dev->tl);
 		unregister_netdev(dev->netdev);
 		unlink_all_urbs(dev);
 		free_all_urbs(dev);
